raster-foundry/raster-foundry
# Download support for uploads

## Overview

This PR adds download support for user uploads. Previously, since the location of user uploads are not public and were S3 protocol URIs, they were not accessible.

This PR adds an endpoint `/api/scenes/<scene_id>/download` that returns all of the images associated with a scene where each image has a `downloadUri`.

In cases where the scene resides in our own infrastructure, a signed URL is generated and used. In case where the scene's `sourceUri` is an `s3://` URI, it is converted to an HTTPS URI. Otherwise, this field ends up being the same as `sourceUri`.

This PR also adds front-end changes to support this new endpoint.

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~


## Testing Instructions

 * Go to imports on the front-end and try downloading the imported scenes
 * You can inspect the network response from the new endpoint an ensure that each image returned has a `downloadUri` property

Closes #2816
